content/en/docs/generating-objects/instantiate-methods.md

```diff
@@ -0,0 +1,644 @@
+---
+title: "Instantiate Methods"
+images: []
+menu:
+docs:
+parent: "generating-objects"
+identifier: "instantiate-methods"
+weight: 32
+---
+
+There are several ways of creating an object. For example, you can initialize an instance of an object using its constructor, or you can use the factory method it has.
+
+Fixture Monkey allows you to choose the preferred method of creating your object through the `instantiate()` method.
+{{< alert icon="💡" text="If the Kotlin Plugin is added, you can use the instantiateBy() method with a custom DSL." />}}
+
+From the `ArbitraryBuilder`, you can determine how the object is created by specifying the preferred instantiation method (constructor or factory method).
+
+This doesn't mean that you have to tell the `ArbitraryBuilder` how to create an object every time you use it.
+If you want to set a global option as the default method for creating all objects with a Fixture Monkey instance, refer to the [Introspector](../introspector) page.
+
+The `instantiate()` method is just a convenient way to modify the generation method from the `ArbitraryBuilder`.
+
+{{< alert icon="💡" text="This feature is still experimental. Currently only constructors and factory methods can be used. More options will be added in the future." />}}
```

Comment:
If there are any missing features related to `instantiate()`, please let me know. 
I would appreciate it if you could also inform me of the additional options that will be added.

Reply:
It is a stable feature now. It is better to remove this alert.

Reply:
Creating an instance by the constructor with a default argument in Kotlin is missing.

Reply:
Removed the alert! I didn't understand the second comment; is it addressing a different feature related to the code below?
```kotlin
@Test
fun test() {
    val product = fixtureMonkey.giveMeBuilder<Product>()
        .instantiateBy {
            constructor()
        }
        .sample()

    then(product.productName).isEqualTo("")
}
```

Reply:
```kotlin
class Product(val productName = "test")

@Test
fun test() {
    val product = fixtureMonkey.giveMeBuilder<Product>()
        .instantiateBy {
            constructor{
                 parameter<String>(useDefaultArgument = true)
            }
        }
        .sample()

    then(product.productName).isEqualTo("test")
}
```